allen_oephys_to_nwb/allen_ecephys_interface.py

```diff
@@ -0,0 +1,205 @@
+from nwb_conversion_tools.basedatainterface import BaseDataInterface
+from nwb_conversion_tools.utils import get_base_schema, get_schema_from_hdmf_class
+from . import schema
+import pynwb
+from pynwb import NWBFile
+from datetime import datetime
+from pathlib import Path
+import importlib.resources as pkg_resources
+import numpy as np
+import h5py
+import json
+import uuid
+
+
+class AllenEcephysInterface(BaseDataInterface):
+
+    @classmethod
+    def get_input_schema(cls):
+        with pkg_resources.open_text(schema, 'source_schema.json') as f:
+            input_schema = json.load(f)
+        return input_schema['properties']
+
+    def __init__(self, **input_args):
+        super().__init__(**input_args)
+
+    def get_metadata_schema(self):
+        metadata_schema = get_base_schema()
+        metadata_schema['properties']['Ecephys'] = get_base_schema(tag='Ecephys')
+        metadata_schema['properties']['Ecephys']['properties']['Device'] = get_schema_from_hdmf_class(pynwb.device.Device)
+        metadata_schema['properties']['Ecephys']['properties']['ElectrodeGroup'] = get_schema_from_hdmf_class(pynwb.ecephys.ElectrodeGroup)
+        metadata_schema['properties']['Ecephys']['properties']['ElectricalSeries_raw'] = get_schema_from_hdmf_class(pynwb.ecephys.ElectricalSeries)
+
+        metadata_schema['properties']['Ophys'] = get_base_schema(tag='Ophys')
+        metadata_schema['properties']['Ophys']['properties']['Device'] = get_schema_from_hdmf_class(pynwb.device.Device)
+        metadata_schema['properties']['Ophys']['properties']['TwoPhotonSeries'] = get_schema_from_hdmf_class(pynwb.ophys.TwoPhotonSeries)
+        metadata_schema['properties']['Ophys']['properties']['Fluorescence'] = get_schema_from_hdmf_class(pynwb.ophys.Fluorescence)
+        metadata_schema['properties']['Ophys']['properties']['ImagingPlane'] = get_schema_from_hdmf_class(pynwb.ophys.ImagingPlane)
+        # metadata_schema['properties']['Ophys']['properties']['ImageSegmentation'] = get_schema_from_hdmf_class(pynwb.ophys.ImageSegmentation)
```

Comment:
Why is there ophys stuff in the Ecephys data interface?

Reply:
it will not be here, this was leftover of some tests I was running